src/server-extension/resolvers/AdminResolver/index.ts

```diff
@@ -65,14 +103,48 @@ export class AdminResolver {
         args.commentsWeight,
         args.reactionsWeight,
         [args.joysteamTimestampSubWeight, args.ytTimestampSubWeight],
+        args.defaultChannelWeight,
       ],
       em
     )
     await videoRelevanceManager.updateVideoRelevanceValue(em, true)
     return { isApplied: true }
   }
 
-  @UseMiddleware(OperatorOnly)
+  @UseMiddleware(OperatorOnly(OperatorPermission.SET_CHANNEL_WEIGHTS))
+  @Mutation(() => [ChannelWeight])
+  async setChannelsWeights(@Args() { inputs }: SetChannelsWeightsArgs): Promise<ChannelWeight[]> {
+    const em = await this.em()
+
+    const results: ChannelWeight[] = []
+
+    // Process each SetChannelWeightInput
+    for (const weightInput of inputs) {
+      const { channelId, weight } = weightInput
+
+      // Update the channel weight in the database
+      const updateResult: UpdateResult = await em.transaction(
```

Comment:
What happens here if the provided `channelId` doesn't exist in the database?

Reply:
Nothing will happen, No data is altered since the `channelId` does not match any existing records. And you will see in the mutation response if the specific update is successful or not (through `isApplied` field)
<img width="873" alt="image" src="https://github.com/Joystream/orion/assets/37098720/17cf3a72-a782-4168-9c51-e5cba6d29be9">